src/plugins/vis_type_timeseries/public/application/components/data_source_picker.tsx

```diff
@@ -0,0 +1,103 @@
+ */
+
+import React, { useEffect, useState } from 'react';
+import _ from 'lodash';
+
+import { SavedObjectsClientContract, ToastsStart } from 'src/core/public';
+import { DataSourceManagementPluginSetup } from 'src/plugins/data_source_management/public';
+import { DATA_SOURCE_ID_KEY } from '../../../common/constants';
+
+/**
+ * Provide only the necessary plugin setup/start and dataSourceId and this component will render a DataSourceSelector with a default selection
+ *
+ * @property {SavedObjectsClientContract} savedObjectsClient
+ * @property {DataSourceManagementPluginSetup} dataSourceManagement
+ * @property {ToastsStart} toasts
+ * @property {string} [defaultDataSourceId] - the datasource id as the default option when the component first renders
+ * @property {(e: Array<{}>) => void} handleChange - the function that will update the model when a datasource is selected
+ */
+export interface DataSourcePickerProps {
+  savedObjectsClient: SavedObjectsClientContract;
+  dataSourceManagement: DataSourceManagementPluginSetup;
+  toasts: ToastsStart;
+  defaultDataSourceId?: string;
+  handleChange: (e: Array<{}>) => void;
+}
+
+/**
+ * Provides a wrapper around the DataSourceSelector component exposed by core
+ *
+ * @param {DataSourcePickerProps} props
+ * @returns the DataSourceSelector
+ */
+export const DataSourcePicker = (props: DataSourcePickerProps) => {
+  const { savedObjectsClient, defaultDataSourceId, handleChange } = props;
+  const [defaultOption, setDefaultOption] = useState<Array<{ id: string; label: string }>>();
+  const DataSourceSelector = props.dataSourceManagement.ui.DataSourceSelector;
+
+  const onDataSourceSelectChange = (dataSourceOption: Array<{ id: string; label: string }>) => {
+    setDefaultOption(dataSourceOption);
+    handleChange(dataSourceOption);
+  };
+
+  useEffect(() => {
+    if (!defaultDataSourceId || defaultDataSourceId === '') {
+      // @ts-expect-error
+      setDefaultOption(null);
+      return;
+    }
+
+    getDataSourceTitleFromId(defaultDataSourceId, savedObjectsClient).then((label) => {
+      if (!!label) {
+        setDefaultOption([
+          {
+            id: defaultDataSourceId,
+            label,
+          },
+        ]);
+        return;
+      }
+      // @ts-expect-error
+      setDefaultOption(null);
+    });
+  }, [defaultDataSourceId, savedObjectsClient]);
+
+  if (defaultOption === undefined) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <DataSourceSelector
+      savedObjectsClient={savedObjectsClient}
+      notifications={props.toasts}
+      onSelectedDataSource={onDataSourceSelectChange}
+      defaultOption={defaultOption}
+      disabled={false}
+      fullWidth={false}
+      removePrepend={true}
+      // @ts-expect-error
+      filterFunc={(ds) => ds.attributes.auth.type !== 'no_auth'}
+    />
+  );
+};
+
+export const createDataSourcePickerHandler = (handleChange: any) => {
+  return (selectedOptions: []): void => {
+    return handleChange?.({
```

Comment:
In which case handleChange would be undefined?

Reply:
I believe `handleChange` came from the parent component. I used a modified `createSelectHandler()` which also assumes `handleChange` can be undefined (`src/plugins/vis_type_timeseries/public/application/components/lib/create_select_handler.js`)